cookbooks/bcpc-hadoop/libraries/utils.rb

```diff
@@ -395,3 +395,33 @@ def has_vip?
   ).run_command
   cmd.stderr.empty? && cmd.stdout.include?(node[:bcpc][:management][:vip])
 end
+
+def oozie_running?(host)
+  begin
+    oozie_url = "sudo -u oozie oozie admin -oozie http://#{host}:11000/oozie -status"
+    cmd = Mixlib::ShellOut.new(
+      oozie_url, :timeout => 10
+    ).run_command
+    Chef::Log.debug("Oozie status: #{cmd.stdout}")
+    cmd.exitstatus == 0 && cmd.stdout.include?('NORMAL')
+  rescue Exception => e
+    Chef::Log.warn("oozie_running? on #{host}: #{e.message}")
+    false
+  end
+end
+
+def update_oozie_sharelib(host)
+  begin
+    if oozie_running?(host) 
+      update_sharelib = "sudo -u oozie oozie admin -oozie http://#{host}:11000/oozie -sharelibupdate"
+      cmd = Mixlib::ShellOut.new(
+        update_sharelib, :timeout => 10
+      ).run_command
+      Chef::Log.info("Sharelibupdate: Updated sharelib on #{host}") if cmd.exitstatus == 0 
+    else
+      Chef::Log.info("Sharelibupdate: Failed to update sharelib on #{host}")
+    end
+  rescue Exception => e
+    Chef::Log.error(e.message)
```

Comment:
I would like us not to swallow exceptions unless we have a good reason for handling them elsewhere.